src/organization/organization.service.ts

```diff
@@ -299,4 +300,58 @@ export class OrganizationService {
     };
     return json?.Data?.[0]?.['subOrgs'] || [];
   }
+
+  private async getOrgDefinitionByUid(uid: string) {
+    const query = `
+    {
+      definitions(func: uid(${uid})) @filter(type(OrgDefinition))
+      ${roleDefinitionFullQuery}
+    }
+    `;
+    const res = await this.dgraph.query(query);
+    const { definitions } = res.getJson() as {
+      definitions: OrganizationDefinitionDTO[];
+    };
+    const definition = definitions[0];
+    if (!definition) {
+      return {
+        uid,
+      } as OrganizationDefinitionDTO;
+    }
+    return definition;
+  }
+
+  private getDefinitionForOrgs(orgs?: Organization[]) {
+    return (
+      orgs &&
+      Promise.all(
+        orgs.map(async org => ({
+          ...org,
+          definition: await this.getOrgDefinitionByUid(org.definition.uid),
```

Comment:
Why do you need to get the OrgDefinition again? It seems like a bunch of the data (ie. uid, name, owner, namespace, definition) is being retrieved already in `getOrganizationNestedSubOrgs`?

Reply:
In the @recurse function we can get only top level props, so the definition looks like this { uuid: string } and we need to get the rest manually :)